File: liqp-core/src/main/java/liqp/filters/Last.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Last extends Filter {

    /*
     * last(array)
     *
     * Get the last element of the passed in array
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        Object[] array = super.asArray(value);

        return array.length == 0 ? null : super.asString(array[array.length - 1]);
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Last.kt
package liqp.filter

import liqp.context.LContext

class Last : LFilter() {

  /**
   * last(array)
   *
   * Get the last element of the passed in array
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asIterable(value).lastOrNull()
  }
}
